samtranslator/model/sam_resources.py

```diff
@@ -526,13 +527,17 @@ def to_cloudformation(self, **kwargs):
                                      resource_attributes=self.resource_attributes,
                                      passthrough_resource_attributes=self.get_passthrough_resource_attributes(),
                                      open_api_version=self.OpenApiVersion,
-                                     models=self.Models)
+                                     models=self.Models,
+                                     domain=self.Domain)
 
-        rest_api, deployment, stage, permissions = api_generator.to_cloudformation()
+        rest_api, deployment, stage, permissions, domain, basepath_mapping = api_generator.to_cloudformation()
 
         resources.extend([rest_api, deployment, stage])
         resources.extend(permissions)
-
+        if domain:
+            resources.extend([domain])
+        if basepath_mapping:
+            resources.extend(basepath_mapping)
```

Comment:
Doesn't it need to be this?

```suggestion
            resources.extend([basepath_mapping])
```

Reply:
It always returns a list